public/client.js

```diff
@@ -0,0 +1,262 @@
+
+var log = console.log.bind(console), keyData = document.getElementById('key_data'),
+		deviceInfoInputs = document.getElementById('inputs'), deviceInfoOutputs = document.getElementById('outputs'), midi;
+var AudioContext = AudioContext || webkitAudioContext; // for ios/safari
+var context = new AudioContext();
+var activeNotes = [];
+var btnBox = document.getElementById('content'), btn = document.getElementsByClassName('button');
+var data, cmd, channel, type, note, velocity;
+
+// request MIDI access
+if(navigator.requestMIDIAccess){
+	navigator.requestMIDIAccess({sysex: false}).then(onMIDISuccess, onMIDIFailure);
+}
+else {
+	alert("No MIDI support in your browser.");
+}
+
+// add event listeners
+document.addEventListener('keydown', keyController);
+document.addEventListener('keyup', keyController);
+for(var i = 0; i < btn.length; i++){
+	btn[i].addEventListener('mousedown', clickPlayOn);
+	btn[i].addEventListener('mouseup', clickPlayOff);
+}
+// prepare audio files
+for(var i = 0; i < btn.length; i++){
+	addAudioProperties(btn[i]);
+}
+
+var sampleMap = {
+	key60: 1,
+	key61: 2,
+	key62: 3,
+	key63: 4,
+	key64: 5
+};
+// user interaction
+function clickPlayOn(e){
+	e.target.classList.add('active');
+	e.target.play();
+}
+
+function clickPlayOff(e){
+	e.target.classList.remove('active');
+}
+
+function keyController(e){
+	if(e.type == "keydown"){
+		switch(e.keyCode){
+			case 81:
+				btn[0].classList.add('active');
```

Comment:
I wonder if the code in these case statements can be extracted so that it'll be clearer what is happening. Such as some 
```
function activateToPlay (buttonIndex) {
  btn[buttonIndex].classList.add('active); 
  btn[buttonIndex].play();
}
```